modules/unsupported/process-raster/src/test/java/org/geotools/process/raster/ContourProcessTest.java

```diff
@@ -244,24 +244,34 @@ private GridCoverage2D createVerticalGradient(
     }
 
     @Test
-    public void testContourWithTransformAndReprojection() throws Exception {
-        int width = 256;
-        int height = 256;
+    public void testContourWithTransformAndReprojection_WithOpacitySingleFTS() throws Exception {
+        // The style with a single FeatureTypeStyle will use plain rendering.
         // The style with opacity will not use the screenmap.
-        int whiteSamples1 = countWhiteSamples("contour_with_opacity.sld", width, height);
+        countWhiteSamples("contour_with_opacity.sld");
+    }
+
+    @Test
+    public void testContourWithTransformAndReprojection_WithoutOpacitySingleFTS() throws Exception {
+        // The style with a single FeatureTypeStyle will use plain rendering.
         // The style without opacity will use the screenmap.
-        int whiteSamples2 = countWhiteSamples("contour_without_opacity.sld", width, height);
-        // The white contour lines with the sample data and styles should cover about
-        // 0.25% of the rendered image regardless of whether the screenmap is used.
-        // The bug that this is testing would cause a large number of diagonal lines
-        // to be drawn across the image so the percentage of white pixels with the
-        // bug would be significantly higher.
-        double area = width * height;
-        assertEquals(0.0025, whiteSamples1 / area, 0.0005);
-        assertEquals(0.0025, whiteSamples2 / area, 0.0005);
+        countWhiteSamples("contour_without_opacity.sld");
+    }
+
+    @Test
+    public void testContourWithTransformAndReprojection_WithOpacityMultiFTS() throws Exception {
+        // The style with multiple FeatureTypeStyles will use optimized rendering.
+        // The style with opacity will not use the screenmap.
+        countWhiteSamples("contour_multi_fts_with_opacity.sld");
     }
 
-    private int countWhiteSamples(String styleName, int width, int height) throws Exception {
+    @Test
+    public void testContourWithTransformAndReprojection_WithoutOpacityMultiFTS() throws Exception {
+        // The style with multiple FeatureTypeStyles will use optimized rendering.
+        // The style without opacity will use the screenmap.
+        countWhiteSamples("contour_multi_fts_without_opacity.sld");
+    }
+
+    private void countWhiteSamples(String styleName) throws Exception {
```

Comment:
Love the reuse...  but the method is now asserting something, not just counting white pixels. This sounds like a better name (open to suggestions):

```suggestion
    private void assertWhiteSamples(String styleName) throws Exception {
```

Reply:
I made this change.